src/Persisters/Flysystem.php

```diff
@@ -162,17 +126,23 @@ public function save(Persistable $persistable) : void
      */
     public function load() : Persistable
     {
-        if (!$this->filesystem->has($this->path)) {
-            throw new RuntimeException("File does not exist in filesystem. {$this}");
+        if (!$this->filesystem->fileExists($this->path)) {
+            throw new RuntimeException("File does not exist at {$this->path}.");
+        }
+
+        try {
+            $data = $this->filesystem->read($this->path);
+        } catch (FilesystemException $e) {
+            throw new RuntimeException("Error reading data from {$this->path}.");
         }
 
-        $data = new Encoding($this->filesystem->read($this->path) ?: '');
+        $encoding = new Encoding($data);
 
-        if ($data->bytes() === 0) {
-            throw new RuntimeException("File does not contain any data. {$this}");
+        if ($encoding->bytes() === 0) {
+            throw new RuntimeException("File at {$this->path} does not contain any data.");
```

Comment:
❓  **Question:** As we are throwing a `RuntimeException` when trying to _read_ an empty encodings, should we also raise a similar exception in the `save` method if trying to _write_ an empty encoding? 🤔

Reply:
Yes, we should probably be consistent, nice catch